### ISSUE TITLE
Support maxProperties (and/or consider for oneOf semantics)

### ISSUE THREAD

**Issue by alechenninger** [swagger-api/swagger-codegen#4307]
##### Description

maxProperties does not appear to be used in swagger codegen.

##### Swagger-codegen version

commit: fdb6d30 (2.2.2-SNAPSHOT)

##### Suggest a Fix

maxProperties could be used to indicate only N fields should be non-null. Where n=1, this is "oneOf" semantics.

For example:

```
MyObject:
  type: object
  properties:
    thing1:
      $ref: '#/definitions/Thing1'
    thing2:
      $ref: '#/definitions/Thing2'
  maxProperties: 1
```

For objects with maxProperties, setters could validate at runtime whether the maxProperties count has been breached. In fact, this is similar to how gRPC generates code for protobuf oneof definitions, which unsets other fields if another among the "oneof" has been set.

**Comment by wing328** [comment]
@alechenninger yup, it's not supported at the moment. We'll try to support it after the 2.2.2 release.

We also welcome PR to support `maxProperties`, `minProperties`

**Comment by ebelanger** [comment]
@wing328 Here is another small contribution for minProperties and maxProperties support by reusing minItems and maxItems so no template change is required.

**Comment by wing328** [comment]
Closed via #5395
